cmd/tailscaled/onboot.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"tailscale.com/version/distro"
+)
+
+var onBootFunc = onBoot // so it can be addressable
+
+func onBoot(args []string) error {
+	if len(args) != 0 {
+		return errors.New("unsupported arguments")
+	}
+	if runtime.GOOS != "linux" || distro.Get() != distro.Synology {
+		return errors.New("only implemented on Synology")
+	}
+	if uid := os.Getuid(); uid != 0 {
+		return fmt.Errorf("must be run as root, not %q (%v)", os.Getenv("USER"), uid)
+	}
+	if _, err := os.Stat("/dev/net/tun"); os.IsNotExist(err) {
+		if err := os.MkdirAll("/dev/net", 0755); err != nil {
```

Comment:
users have reported that the permissions can be wrong, so do the `chmod` outside of the `IsNotExist` check

Reply:
done